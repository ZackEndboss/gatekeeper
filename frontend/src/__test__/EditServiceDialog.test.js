import React from 'react';
import renderer from 'react-test-renderer';
import {MockStore} from './mockStore';
import {Provider} from 'react-redux';
import {Modal} from 'react-bootstrap';
import {shallow, mount} from 'enzyme';
import ConnectedEditServiceDialog, {EditServiceDialog, ServiceForm}  from '../EditServiceDialog';

jest.mock('../FormInputField', () => 'FormInputField');

describe('EditServiceDialog', () => {
    it('renders', () => {
        const wrapper = shallow(<EditServiceDialog service='service'
                                                   group='group'
                                                   environments='env1, env2, env3'
                                                   show={true}
                                                   setService='setService'
                                                   closeEditServiceDialog='closeEditServiceDialog'/>);
        expect(wrapper.find(Modal).length).toEqual(1);
    });

    it('dialog is open', () => {
        const wrapper = shallow(<EditServiceDialog service='service'
                                                   group='group'
                                                   environments='env1, env2, env3'
                                                   show={true}
                                                   setService='setService'
                                                   closeEditServiceDialog='closeEditServiceDialog'/>);
        expect(wrapper.find(Modal).prop('show')).toEqual(true);
    });

    it('contact form should render form corretly', () => {
        const component = renderer.create(<ServiceForm comment='comment 1'/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});

describe('load from state', () => {
    it('should load the state correctly', () => {
        jest.resetModules();
        const storeMock = MockStore({
            dialog: {
                editService: {
                    group: 'ftX',
                    service: 'gatekeeper',
                    show: true
                }
            }
        });

        const subject = mount(
            <Provider store={storeMock}>
                <ConnectedEditServiceDialog/>
            </Provider>);
        expect(subject.find(Modal).prop('show')).toEqual(true);
    });
});