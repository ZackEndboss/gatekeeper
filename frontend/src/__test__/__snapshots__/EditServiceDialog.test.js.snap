// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditServiceDialog contact form should render form corretly 1`] = `
<form
  onSubmit={undefined}
>
  <FormInputField
    inputProps={
      Object {
        "autoFocus": true,
        "component": "input",
        "type": "text",
      }
    }
    label="Name"
    name="name"
  />
  <FormInputField
    inputProps={
      Object {
        "component": "input",
        "type": "text",
      }
    }
    label="Group"
    name="group"
  />
  <FormInputField
    inputProps={
      Object {
        "component": "input",
        "type": "text",
      }
    }
    label="Environments"
    name="environments"
    placeholder="Env1, Env2, Env3"
  />
  <button
    className="btn btn-danger"
    disabled={false}
    onClick={undefined}
    type="button"
  >
    Delete
  </button>
  <div
    style={
      Object {
        "float": "right",
      }
    }
  >
    <button
      className="btn btn-success"
      disabled={false}
      type="submit"
    >
      Save
    </button>
  </div>
</form>
`;
